### PR TITLE
Feature/1090 list clusters

### PR DESCRIPTION
**Pull request by mvcninja**
- Add cluster node class / unit test
- Implement retrieval of regional clusters

### CODE REVIEW

File: packages/core/src/shared/clients/docdbClient.ts [hunk 1]
```diff
@@ -3,14 +3,33 @@
...
 export class DefaultDocumentDBClient {
     public constructor(public readonly regionCode: string) {}
+
+    public async getClient(): Promise<DocDBClient> {
+        const credentials = await globals.awsContext.getCredentials()
+        const config: DocDBClientConfig = {
+            credentials: credentials,
+            region: this.regionCode,
+        }
+        return new DocDBClient(config)
```
Comment (alexey-temnikov) [inline packages/core/src/shared/clients/docdbClient.ts:25]: Shall we do an error handling here, ensuring the client can be created and credentials and regions are populated?

File: packages/core/src/shared/clients/docdbClient.ts [hunk 2]
```diff
@@ -3,14 +3,33 @@
...
+    public async listClusters(): Promise<DBCluster[]> {
+        const client = await this.getClient()
+        const input = {
+            Filters: [{ Name: 'engine', Values: ['docdb'] }],
+        }
+
+        const command = new DescribeDBClustersCommand(input)
+        const response = await client.send(command)
+
+        return response.DBClusters ?? []
```
Comment (alexey-temnikov) [inline packages/core/src/shared/clients/docdbClient.ts]: Error handling - if unable to retrieve information.

File: packages/core/src/docdb/explorer/dbClusterNode.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { inspect } from 'util'
+import { DBCluster } from '@aws-sdk/client-docdb'
+import { AWSTreeNodeBase } from '../../shared/treeview/nodes/awsTreeNodeBase'
+
+export class DBClusterNode extends AWSTreeNodeBase {
```
Comment (alexey-temnikov) [inline packages/core/src/docdb/explorer/dbClusterNode.ts:16]: nit: let's add jsDocs with description and `@returns` values

File: packages/core/src/shared/clients/docdbClient.ts (hunk 1 above)
Comment (vkagamlyk) [reply packages/core/src/shared/clients/docdbClient.ts:25]: Error handling is inconsistent over this toolkit.
Options to consider
log error and re-throw same exception (LambdaClient is example) 
throw `ToolkitError` (ec2Client is example)
do not handle exceptions (ecsClient)

I think best option to log error and throw `ToolkitError`

File: packages/core/src/docdb/explorer/docdbNode.ts [hunk 4]
```diff
@@ -24,10 +26,11 @@ export class DocumentDBNode extends AWSTreeNodeBase {
...
         return await makeChildrenNodes({
             getChildNodes: async () => {
-                return Promise.resolve([]) //TODO: Get clusters from region
+                const clusters = await this.client.listClusters()
+                return clusters.map(cluster => new DBClusterNode(cluster))
             },
-            getNoChildrenPlaceholderNode: async () => new PlaceholderNode(this, '[Nothing to see here yet.]'),
-            //new PlaceholderNode(this, localize('AWS.explorerNode.docdb.noClusters', '[No Clusters found]')),
+            getNoChildrenPlaceholderNode: async () =>
+                new PlaceholderNode(this, localize('AWS.explorerNode.docdb.noClusters', '[No Clusters found]')),
```
Comment (alexey-temnikov) [inline packages/core/src/docdb/explorer/docdbNode.ts:33]: Do we need to add any resources with the actual localization?

Comment (mvcninja) [reply packages/core/src/docdb/explorer/docdbNode.ts:33]: Good question to be answered by AWS.

File: packages/core/src/shared/clients/docdbClient.ts [hunk 5]
```diff
@@ -3,14 +3,41 @@
...
+            const input = {
+                Filters: [{ Name: 'engine', Values: ['docdb'] }],
+            }
+            const command = new DescribeDBClustersCommand(input)
+            const response = await client.send(command)
+            return response.DBClusters ?? []
+        } catch (e) {
+            throw ToolkitError.chain(e, 'Failed to get DocumentDB clusters')
+        }
+    }
```
Comment (kmcginnes) [inline packages/core/src/shared/clients/docdbClient.ts:42]: This logic seems like it should be testable and would benefit from tests. Is it possible to restructure to allow testing of this logic?

Comment (mvcninja) [reply packages/core/src/shared/clients/docdbClient.ts:42]: This is just a lightweight wrapper around the v3 sdk client, not sure how much benefit tests would add here.

Comment (kmcginnes) [reply packages/core/src/shared/clients/docdbClient.ts:42]: You might be right. I'm not saying it is necessary, just something I noticed. 

Here are the questions that can be answered by testing, if we wanted to go this far:

- Does `listClusters()` send the correct command with the right parameters to the client?
- What happens if the client responds with `DBClusters === null || DBClusters === undefined`?
- What happens if the client throws an error?
    - Is the error the right type?
    - Does the error contain the original thrown error?

**Pull request merged**